api/v1alpha1/component_types.go

```diff
@@ -62,103 +56,27 @@ type ComponentSpec struct {
 	Secret string `json:"secret,omitempty"`
 
 	// Source describes the Component source
-	Source ComponentSource `json:"source,omitempty"`
+	Source *ComponentSource `json:"source,omitempty"`
```

Comment:
how about 

```suggestion
	GitSource *GitSource `json:"gitSource,omitempty"`
```

because `ContainerImage` can also be a source.  This would mean also renaming `type ComponentSource struct{}`

I am thinking about making this change in the application-service too after thinking about it now. it looks cleaner